src/rdkafka_partition.c

```diff
@@ -1959,7 +1966,7 @@ rd_kafka_resp_err_t rd_kafka_toppar_op_fetch_start (rd_kafka_toppar_t *rktp,
                                                     rd_kafka_replyq_t replyq) {
 	int32_t version;
 
-        if (fwdq)
+        if (fwdq && !rd_kafka_toppar_fwd_app(rktp))
                 rd_kafka_q_fwd_set(rktp->rktp_fetchq, fwdq);
```

Comment:
this should be done in toppar_fetch_start(),  this function is just the thread-agnostic caller.
And it should hold the lock outside the is_fwd_app() checker too to avoid races.